components/concept/push/src/main/java/mozilla/components/concept/push/PushService.kt

```diff
@@ -32,4 +33,17 @@ interface PushService {
      * Forces the push service to renew it's registration. This may lead to a new registration token being received.
      */
     fun forceRegistrationRenewal()
+
+    /**
+     * Checks if the messaging service is available to use.
+     */
+    fun isAvailable(context: Context): Boolean
+}
+
+/**
+ * Supported push services.
+ */
+enum class ServiceType {
+    FCM,
+    ADM
```

Comment:
This being an enum in the `concept` significantly limits the flexibility of this concept? 🤔

Reply:
That's a good point. The `ServiceType` and `PushType` can now be moved further up to the feature since we're integrating the rust component at that level.